extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -81,101 +87,157 @@ public SecurityIdentity get() {
+        return userInfo.onItem().transformToUni(
+                new Function<JsonObject, Uni<? extends SecurityIdentity>>() {
+                    @Override
+                    public Uni<SecurityIdentity> apply(JsonObject userInfo) {
+                        return createSecurityIdentityWithOidcServerUni(vertxContext, request, resolvedContext, userInfo);
+                    }
+                });
+    }
+
+    private Uni<SecurityIdentity> createSecurityIdentityWithOidcServerUni(RoutingContext vertxContext,
+            TokenAuthenticationRequest request, TenantConfigContext resolvedContext, final JsonObject userInfo) {
 
-                resolvedContext.auth.decodeToken(request.getToken().getToken(),
-                        new Handler<AsyncResult<AccessToken>>() {
+        final TokenCredential tokenCred = request.getToken();
+        if (tokenCred instanceof AccessTokenCredential && ((AccessTokenCredential) tokenCred).isOpaque()) {
+            // remote introspection is required, a blocking call
+            if (BlockingOperationControl.isBlockingAllowed()) {
+                return Uni.createFrom().emitter(new Consumer<UniEmitter<? super SecurityIdentity>>() {
+                    @Override
+                    public void accept(UniEmitter<? super SecurityIdentity> uniEmitter) {
+                        createSecurityIdentityWithOidcServer(uniEmitter, vertxContext, request, resolvedContext, userInfo);
+                    }
+                });
+            } else {
+                return Uni.createFrom().emitter(
+                        new Consumer<UniEmitter<? super SecurityIdentity>>() {
                             @Override
-                            public void handle(AsyncResult<AccessToken> event) {
-                                if (event.failed()) {
-                                    uniEmitter.fail(new AuthenticationFailedException(event.cause()));
-                                    return;
-                                }
+                            public void accept(UniEmitter<? super SecurityIdentity> uniEmitter) {
+                                tenantResolver.getBlockingExecutor().execute(new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        createSecurityIdentityWithOidcServer(uniEmitter, vertxContext, request, resolvedContext,
+                                                userInfo);
+                                    }
+                                });
+                            }
+                        });
+            }
+        } else {
+            return Uni.createFrom().emitter(new Consumer<UniEmitter<? super SecurityIdentity>>() {
+                @Override
+                public void accept(UniEmitter<? super SecurityIdentity> uniEmitter) {
+                    createSecurityIdentityWithOidcServer(uniEmitter, vertxContext, request, resolvedContext, userInfo);
```

Comment:
if I'm not mistaken this may be blocking, here you may be on the event loop.

Reply:
@cescoffier please see [this comment](https://github.com/quarkusio/quarkus/pull/13257/files#r522163580) below. Unless it is an opaque token it should be fine to have on the IO thread, because it is a JWT token which is verified by a locally cached JWK keys